src/file_pipe_log/pipe_builder.rs

```diff
@@ -16,12 +16,12 @@ use crate::config::{Config, RecoveryMode};
 use crate::env::FileSystem;
 use crate::event_listener::EventListener;
 use crate::log_batch::LogItemBatch;
-use crate::pipe_log::{FileId, FileSeq, LogQueue};
+use crate::pipe_log::{FileId, LogQueue};
 use crate::util::Factory;
 use crate::{Error, Result};
 
-use super::format::{lock_file_path, FileNameExt, LogFileHeader};
-use super::log_file::build_file_reader;
+use super::format::{lock_file_path, FileNameExt, LogFileHeader, Version};
+use super::log_file::{build_file_reader, FileHandler};
```

Comment:
I don't like reusing it. The version field in `FileHandler` doesn't have to be `Option`. Make it optional is confusing.

Reply:
The reason why it defined with`Option` is that i wanna distinguish whether the `version` is initiated or not. At the begging, it should be `None` when the `SinglePipe` is not recovered.

Reply:
This one is not addressed. I believe `Pipe` is of higher priority here. It creates two issues: (1) adding seq to pipe's fileinfo creates multiple sources of truth; (2) using optional version doesn't convey the true meaning of the data (the version is already validated during recovery).